Block/CategoryMapping.php

```diff
@@ -63,50 +59,44 @@ class CategoryMapping extends Template
     /** @var NostoHelperData */
     private $nostoHelperData;
 
+    /** @var NostoHelperScope */
+    private $nostoHelperScope;
+
     /** @var Logger */
     private $logger;
 
     /**
      * CategoryMapping constructor.
-     * @param StoreManagerInterface $storeManager
      * @param CollectionFactory $collectionFactory
      * @param CategoryBuilder $categoryBuilder
      * @param NostoHelperData $nostoHelperData
+     * @param NostoHelperScope $nostoHelperScope
      * @param Context $context
      * @param Logger $logger
      */
     public function __construct(
-        StoreManagerInterface $storeManager,
         CollectionFactory $collectionFactory,
         CategoryBuilder $categoryBuilder,
         NostoHelperData $nostoHelperData,
+        NostoHelperScope $nostoHelperScope,
         Context $context,
         Logger $logger
     ) {
         parent::__construct($context);
-        $this->storeManager = $storeManager;
         $this->collectionFactory = $collectionFactory;
         $this->categoryBuilder = $categoryBuilder;
         $this->nostoHelperData = $nostoHelperData;
+        $this->nostoHelperScope = $nostoHelperScope;
         $this->logger = $logger;
     }
 
     /**
-     * @return false|string
+     * @return string
      */
     public function getCategoryMap()
     {
-
-        $array = [];
-        try {
-            $store = $this->storeManager->getStore();
-            if ($store instanceof Store) {
-                $array = $this->getMagentoCategories($store);
-            }
-        } catch (NoSuchEntityException $e) {
-            $this->logger->exception($e);
-        }
-
+        $store = $this->nostoHelperScope->getStore();
```

Comment:
I don't think the `storeId` is available here so let's add a comement.

Reply:
It fetches current store if no storeId is given. What comment should I leave?

Reply:
Yes, but if there's the possibility to pass the store id then we should pass it. If there's no way to get the store id, just put a comment.